### PR TITLE
Add common hex encode/decode fuzz tests

### PR DESCRIPTION
**Pull request by ericlin0101**
Signed-off-by: Lin Lin <linlineric@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: test/common/common/hex_fuzz_test.cc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#include "common/common/hex.h"
+
+#include "test/fuzz/fuzz_runner.h"
+
+namespace Envoy {
+namespace Fuzz {
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  Envoy::Hex::encode(buf, len);
```
Comment (asraa) [inline test/common/common/hex_fuzz_test.cc]: A hex encode/decode would work for a round-trip validation.

File: test/common/common/hex_fuzz_test.cc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#include "common/common/hex.h"
+
+#include "test/fuzz/fuzz_runner.h"
+
+namespace Envoy {
+namespace Fuzz {
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
```
Comment (asraa) [inline test/common/common/hex_fuzz_test.cc]: Would you be able to also include coverage over `uint64ToHex` and `uint32ToHex`? (An easy way to get ints from raw buffers is to use `FuzzedDataProvider::ConsumeIntegral<uint32>`) 

See https://github.com/google/fuzzing/blob/master/docs/split-inputs.md#fuzzed-data-provider
https://github.com/envoyproxy/envoy/blob/3662462563f4b98fde7935215e551582f7e4ea90/test/common/compressor/compressor_fuzz_test.cc

Comment (ericlin0101) [reply test/common/common/hex_fuzz_test.cc]: Thanks for the information.
Tests added.

File: test/common/common/hex_fuzz_test.cc (hunk 1 above)
Comment (ericlin0101) [reply test/common/common/hex_fuzz_test.cc]: Done

File: test/common/common/hex_fuzz_test.cc [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  Envoy::Hex::encode(buf, len);
+  Envoy::Hex::decode(std::string(reinterpret_cast<const char*>(buf), len));
+
+  FuzzedDataProvider fuzz_data(buf, len);
+  auto decoded = Envoy::Hex::decode(Envoy::Hex::encode(buf, len));
+  FUZZ_ASSERT(fuzz_data.ConsumeBytes<uint8_t>(len) == decoded);
```
Comment (asraa) [inline test/common/common/hex_fuzz_test.cc]: I think calling `ConsumeBytes` with `len` will read off all `len` bytes of the input string, leaving 0 for the integer calls. (https://github.com/llvm/llvm-project/blob/c494ff05ed59712ef90ffb66bfade6bac45a0ab1/compiler-rt/include/fuzzer/FuzzedDataProvider.h#L155)

You might just want to instantiate/use the FuzzedDataProvider after this, or only read some smaller random number of bytes for the encode/decode round-trip test.

Comment (ericlin0101) [reply test/common/common/hex_fuzz_test.cc]: Done. Created FuzzedDataProvider instance to void chopping.

File: test/common/common/hex_fuzz_test.cc [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+
+  FuzzedDataProvider fuzz_data(buf, len);
+  Envoy::Hex::uint64ToHex(fuzz_data.ConsumeIntegral<uint64_t>());
+  Envoy::Hex::uint32ToHex(fuzz_data.ConsumeIntegral<uint32_t>());
+
+  Envoy::Hex::encode(buf, len);
+  Envoy::Hex::decode(std::string(reinterpret_cast<const char*>(buf), len));
+
+  FuzzedDataProvider fuzz_codec_data(buf, len);
```
Comment (asraa) [inline test/common/common/hex_fuzz_test.cc]: Sorry for being nit-picky here, but if you consume the whole buffer for the round-trip, you don't need the provider and can just `FUZZ_ASSERT(std::vector<uint8_t>(buf, buf+len) == decoded`

**Pull request closed without merging**